tools/azsdk-cli/docs/cli-commands-guidelines.md

```diff
@@ -0,0 +1,109 @@
+# Azure SDK Tools CLI Command Guidelines
+
+This document provides comprehensive guidelines for creating CLI commands in the `Azure.Sdk.Tools.Cli` project. All CLI commands must follow these established patterns and conventions to ensure consistency, maintainability, and proper integration with both CLI and MCP (Model Context Protocol) server functionality.
+Azure SDK developers and service team can use these CLI commands to generate, build, test and release Azure SDK.
+
+## Namespace Organization
+
+### Directory Structure
+
+```
+Tools/
+├── Package/          # Package-level operations 
+├── ReleasePlan/      # Release planning operations
+├── TypeSpec/         # TypeSpec operations
+```
+
+## Command Hierarchy
+
+All CLI commands must follow a predefined top-level command hierarchy. Commands are organized into the following categories:
+
+### 1. **package** - Package Operations
+
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.Package`  
+**Command Group:** `SharedCommandGroups.Package`  
+**Verb:** `package`  or `pkg`
+
+For operations at the SDK package level. The package group has further sub-grouping for better organization:
+
+#### Core Package Operations:
+
+- Build/compile SDK code
+- Generate SDKs
+- Release packages
+- Update version information
+- Validate packages
+
+#### Sub-Groups:
+
+##### **readme** - README Operations
+ 
+For generating and updating README files:
+
+- Generate README file
```

Comment:
We want to be consistent with the language we use in our tools, so the user understand we are talking about the same thing. -> this should be a guideline where terminology means the same across tooling and it is used the same too.
in this case, the `content` and the `file` are a specific implementation detail user doesn't need to think too much about. at the end they just want the README.